ai_eval/coach.py

```diff
@@ -0,0 +1,817 @@
+    )
+
+    initial_message = String(
+        display_name=_("Initial message"),
+        default="",
+        scope=Scope.settings,
+    )
+
+    scenario_data = Dict(
+        display_name=_("Scenario data"),
+        help=_("Arbitrary data accessible in prompt templates"),
+        default={
+            "case_details": "",
+            "learning_objectives": [],
+            "evaluation_criteria": [],
+        },
+        scope=Scope.settings,
+    )
+
+    workspace_title = String(
+        display_name=_("Workspace title"),
+        default=_("Add your answer"),
+        scope=Scope.settings,
+    )
+
+    coach_title = String(
+        display_name=_("Coach title"),
+        default=_("Tutor"),
+        scope=Scope.settings,
+    )
+
+    intro_text = String(
+        display_name=_("Introductory text"),
+        help=_("Optional introductory paragraph shown above the chat panes."),
+        default="",
+        scope=Scope.settings,
+        multiline_editor=True,
+    )
+
+    character_1_avatar = String(
+        display_name=_("Character #1 avatar URL"),
+        help=_("URL for character #1 avatar image"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    character_2_avatar = String(
+        display_name=_("Character #2 avatar URL"),
+        help=_("URL for character #2 avatar image"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    character_1_name = String(
+        display_name=_("Character #1 name"),
+        help=_("Name of character #1"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    character_1_role = String(
+        display_name=_("Character #1 role"),
+        help=_("Role of character #1"),
+        scope=Scope.settings,
+        default="Main character",
+    )
+
+    character_1_prompt = String(
+        display_name=_("Character #1 prompt"),
+        help=_("Prompt to instruct the AI model to act as character #1"),
+        scope=Scope.settings,
+        default=SAMPLE_CHARACTER_PROMPT,
+    )
+
+    character_2_name = String(
+        display_name=_("Character #2 name"),
+        help=_("Name of character #2"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    character_2_role = String(
+        display_name=_("Character #2 role"),
+        help=_("Role of character #2"),
+        scope=Scope.settings,
+        default="Coach",
+    )
+
+    character_2_prompt = String(
+        display_name=_("Character #2 prompt"),
+        help=_("Prompt to instruct the AI model to act as character #2"),
+        scope=Scope.settings,
+        default=SAMPLE_CHARACTER_PROMPT,
+    )
+
+    allow_reset = Boolean(
+        display_name=_("Allow reset"),
+        help=_("Allow the learner to reset the chat"),
+        scope=Scope.settings,
+        default=True,
+    )
+
+    conversation_format = String(
+        display_name=_("Conversation format template"),
+        help=_(
+            "Template used to format the conversation, appended to all prompts"
```

Comment:
This needs docs:
- what templating is used?
- link to templating docs
- what does "all prompts" mean? where is this used?
- What format should this be? XML?